src/sched/lthread.c

```diff
@@ -461,6 +487,50 @@ void _lthread_free(struct lthread* lt)
     lt = 0;
 }
 
+int __init_utp(void *p, int set_tp)
+{
+	struct lthread_tcb_base *tcb = (struct lthread_tcb_base *)p;
+	tcb->self = p;
+	tcb->schedctx = __scheduler_self();
+	if (libc.user_tls_enabled && set_tp)
```

Comment:
The first of these checks should be removed.  We want to set up a TLS area with just a stack canary for threads that are not created with `lthread_create_primitive`.

Reply:
`lthread_create` calls this function with set_tp set to 0. Should we remove the entire block?